Edits to ERC (#4)
safe-research/ERCs: ERCS/erc-7955.md

#### 1. Nick's Method

Use Nick's method to randomly generate a signature for a transaction **without** [EIP-155](./eip-155.md) replay protection that deploys the CREATE2 factory. Nick's method ensures that there is no known private key for the account that deploys the CREATE2 factory, meaning that the resulting contract will have a deterministic address and code on all chains. This strategy is used by the _Deterministic Deployment Proxy_ (deployed to `0x4e59b44847b379578588920ca78fbf26c0b4956c`, including on Ethereum Mainnet), one of the most widely used CREATE2 factory contracts.

**Downsides**:

- It does not work on chains that only accept EIP-155 replay-protected transactions.
- It is sensitive to changes in gas parameters on the target chain since the gas price and limit in the deployment transaction is sealed, and a new one cannot be signed without a private key.

> - Reverts, due to alternative gas schedules for example, make the CREATE2 factory no longer deployable.

Reviewer: This change feels more difficult to comprehend. Alternative suggestion:

```suggestion
- Reverts, such as those caused by alternative gas schedules, make the CREATE2 factory no longer deployable.
```

**Downsides**:

- It is permissioned: the party that holds the secret key has the ultimate say on which chains will get the CREATE2 factory deployments.
- This requires carefully guarding a secret key; if it is exposed or lost, deployments are no longer guaranteed on new chains.
- If the transaction at the given nonce is not a successful CREATE2 factory deployment, then it is no longer possible to have a CREATE2 factory at the canonical address; this can happen by human error, for example.

#### 3. Preinstalls

> Have popular CREATE2 deployment factories deployed on new chains by default. This is, for example, what OP Stack and ZKsync do as part of their preinstalls, including the CREATE2 factory contracts mentioned above. This ensures that the CREATE2 factory contracts have known addresses and codes.

Reviewer 2: I think Arbitrum and Polygon do this too, but best check with @Fbartoli

Author: > Arbitrum

Doing a quick search in the arbitrum repositories, it does not _appear_ to be a preinstall (some hits, but it appears to use the contract extensively, but not deploy it as part of their initial state <https://github.com/search?q=org%3AOffchainLabs+0x4e59b44847b379578588920ca78fbf26c0b4956c&type=code> - and it appears that they just deploy it "normally" as part of their bootstrapping: <https://github.com/OffchainLabs/nitro-contracts/blob/0b8c04e8f5f66fe6678a4f53aa15f23da417260e/src/rollup/DeployHelper.sol#L13>). Take this with a grain of salt though - this is just what I found in a few minutes and didn't dig deep to confirm it is correct. I will wait for @Fbartoli to shed some light.

> Polygon

And looking very unlikely for the Polygon (0 hits: <https://github.com/search?q=org%3Amaticnetwork%200x4e59b44847b379578588920ca78fbf26c0b4956c&type=code>)
